src/sources/aws_s3/mod.rs

```diff
@@ -113,6 +117,23 @@ pub struct AwsS3Config {
     #[configurable(metadata(docs::hidden))]
     #[serde(default)]
     log_namespace: Option<bool>,
+
+    #[configurable(derived)]
+    #[serde(default = "default_framing")]
+    #[derivative(Default(value = "default_framing()"))]
```

Comment:
I'm not used to seeing both serde defaults and derivative defaults used together. Is there are reason to have both?

Reply:
I was following the convention from the SQS source. I assume `derivative` is when using `Default::default`, and the serde default is when deserializing. I'm not sure if serde will re-use the existing default or not, and since we are already doing this in other components I was going to keep it the same.

Reply:
Correct. Serde's field-level `default` attribute uses the field's type's default rather than the struct-level default.